packages/react/src/utils/InternalBackdrop.tsx

```diff
@@ -1,29 +1,59 @@
 import * as React from 'react';
-import PropTypes from 'prop-types';
+import PropTypes, { elementType } from 'prop-types';
 
 /**
  * @ignore - internal component.
  */
 const InternalBackdrop = React.forwardRef(function InternalBackdrop(
-  props: React.ComponentPropsWithoutRef<'div'>,
+  props: InternalBackdrop.Props,
   ref: React.ForwardedRef<HTMLDivElement>,
 ) {
+  const { cutout, ...otherProps } = props;
+
+  let clipPath: string | undefined;
+  if (cutout) {
+    const rect = cutout?.getBoundingClientRect();
+    clipPath = `polygon(
+      0% 0%,
+      100% 0%,
+      100% 100%,
+      0% 100%,
+      0% 0%,
+      ${rect.left}px ${rect.top}px,
+      ${rect.left}px ${rect.bottom}px,
+      ${rect.right}px ${rect.bottom}px,
+      ${rect.right}px ${rect.top}px,
+      ${rect.left}px ${rect.top}px
+    )`;
+  }
```

Comment:
This is cool! I remember we discussed modality using backdrops while excluding certain elements, but can this support multiple "holes"?

Reply:
This implementation assumes just one "hole" but it can easily be expanded to support more. It may be more tricky if elements overlap (due to clip-path fill rules), but doable with some maths.